app/controllers/works_controller.rb

```diff
@@ -0,0 +1,81 @@
+class WorksController < ApplicationController
+  
+  def index
+    @works = Work.all
+    @album_collection = Work.where(category: "album")
+    @movie_collection = Work.where(category: "movie")
+    @book_collection = Work.where(category: "book")
+  end
+
+  def new 
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new work_params
+
+    if @work.save
+      redirect_to work_path(@work.id)
+      flash[:success] = "Work Successfully Created"
+    else
+      flash.now[:error] = "Failed to Create Work"
+      render :new, status: :bad_request
+    end
+  end 
+
+  def show
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
+    if @work.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def edit
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
```

Comment:
You have a few methods that require a specific work to be fetched, making this a good place to use controller filters.